### PR TITLE
Add table gcp_bigquery_job. Closes #241

### LINKED ISSUE #241: Add table gcp_bigquery_jobs
**References**
https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/bigquery_job

https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs

Need to validate below scenario:

Use the maximum bytes billed setting to limit query costs.

You can limit the number of bytes billed for a query using the maximum bytes billed setting. When you set maximum bytes billed, the number of bytes that the query will read is estimated before the query execution. If the number of estimated bytes is beyond the limit, then the query fails without incurring a charge.

### PR DESCRIPTION
**Pull request by bigdatasourav**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
SETUP: tests/gcp_bigquery_job []

PRETEST: tests/gcp_bigquery_job

TEST: tests/gcp_bigquery_job
Running terraform
google_bigquery_dataset.named_test_resource: Creating...
google_bigquery_dataset.named_test_resource: Creation complete after 3s [id=projects/pikachu-aaa/datasets/turbottest24659]
google_bigquery_table.named_test_resource: Creating...
google_bigquery_table.named_test_resource: Creation complete after 1s [id=projects/pikachu-aaa/datasets/turbottest24659/tables/turbottest24659]
google_bigquery_job.named_test_resource: Creating...
google_bigquery_job.named_test_resource: Creation complete after 2s [id=projects/pikachu-aaa/jobs/turbottest24659]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 3 added, 0 changed, 0 destroyed.

Outputs:

project_id = "pikachu-aaa"
region_id = "US"
resource_aka = "gcp://bigquery.googleapis.com/projects/pikachu-aaa/jobs/turbottest24659"
resource_id = "pikachu-aaa:US.turbottest24659"
resource_name = "turbottest24659"

Running SQL query: test-get-query.sql
[
  {
    "id": "pikachu-aaa:US.turbottest24659",
    "job_id": "turbottest24659",
    "kind": "bigquery#job",
    "location": "US",
    "project": "pikachu-aaa"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "id": "pikachu-aaa:US.turbottest24659",
    "job_id": "turbottest24659",
    "kind": "bigquery#job"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "gcp://bigquery.googleapis.com/projects/pikachu-aaa/jobs/turbottest24659"
    ],
    "title": "turbottest24659"
  }
]
✔ PASSED

POSTTEST: tests/gcp_bigquery_job

TEARDOWN: tests/gcp_bigquery_job

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select
  job_id,
  self_link,
  creation_time,
  location
from
  gcp_bigquery_job;
+-----------------+---------------------------------------------------------------------------------------------------+---------------------+----------+
| job_id          | self_link                                                                                         | creation_time       | location |
+-----------------+---------------------------------------------------------------------------------------------------+---------------------+----------+
| turbottest76164 | https://bigquery.googleapis.com/bigquery/v2/projects/pikachu-aaa/jobs/turbottest76164?location=US | 2021-06-25 07:46:57 | US       |
| turbottest24659 | https://bigquery.googleapis.com/bigquery/v2/projects/pikachu-aaa/jobs/turbottest24659?location=US | 2021-06-25 09:56:27 | US       |
| turbottest49597 | https://bigquery.googleapis.com/bigquery/v2/projects/pikachu-aaa/jobs/turbottest49597?location=US | 2021-06-25 08:55:46 | US       |
| turbottest36667 | https://bigquery.googleapis.com/bigquery/v2/projects/pikachu-aaa/jobs/turbottest36667?location=US | 2021-06-25 09:06:33 | US       |
| job1            | https://bigquery.googleapis.com/bigquery/v2/projects/pikachu-aaa/jobs/job1?location=US            | 2021-06-25 07:39:59 | US       |
| turbottest56089 | https://bigquery.googleapis.com/bigquery/v2/projects/pikachu-aaa/jobs/turbottest56089?location=US | 2021-06-25 09:07:30 | US       |
| turbottest66929 | https://bigquery.googleapis.com/bigquery/v2/projects/pikachu-aaa/jobs/turbottest66929?location=US | 2021-06-25 09:08:35 | US       |
| turbottest20237 | https://bigquery.googleapis.com/bigquery/v2/projects/pikachu-aaa/jobs/turbottest20237?location=US | 2021-06-25 07:41:52 | US       |
| turbottest39458 | https://bigquery.googleapis.com/bigquery/v2/projects/pikachu-aaa/jobs/turbottest39458?location=US | 2021-06-25 07:50:10 | US       |
+-----------------+---------------------------------------------------------------------------------------------------+---------------------+----------+
> select
  job_id,
  self_link,
  creation_time,
  location
from
  gcp_bigquery_job
where
  state = 'RUNNING';
+--------+-----------+---------------+----------+
| job_id | self_link | creation_time | location |
+--------+-----------+---------------+----------+
+--------+-----------+---------------+----------+
> select
  job_id,
  self_link,
  creation_time,
  location
from
  gcp_bigquery_job
where
  state = 'DONE';
+-----------------+---------------------------------------------------------------------------------------------------+---------------------+----------+
| job_id          | self_link                                                                                         | creation_time       | location |
+-----------------+---------------------------------------------------------------------------------------------------+---------------------+----------+
| job1            | https://bigquery.googleapis.com/bigquery/v2/projects/pikachu-aaa/jobs/job1?location=US            | 2021-06-25 07:39:59 | US       |
| turbottest66929 | https://bigquery.googleapis.com/bigquery/v2/projects/pikachu-aaa/jobs/turbottest66929?location=US | 2021-06-25 09:08:35 | US       |
| turbottest49597 | https://bigquery.googleapis.com/bigquery/v2/projects/pikachu-aaa/jobs/turbottest49597?location=US | 2021-06-25 08:55:46 | US       |
| turbottest24659 | https://bigquery.googleapis.com/bigquery/v2/projects/pikachu-aaa/jobs/turbottest24659?location=US | 2021-06-25 09:56:27 | US       |
| turbottest36667 | https://bigquery.googleapis.com/bigquery/v2/projects/pikachu-aaa/jobs/turbottest36667?location=US | 2021-06-25 09:06:33 | US       |
| turbottest56089 | https://bigquery.googleapis.com/bigquery/v2/projects/pikachu-aaa/jobs/turbottest56089?location=US | 2021-06-25 09:07:30 | US       |
| turbottest39458 | https://bigquery.googleapis.com/bigquery/v2/projects/pikachu-aaa/jobs/turbottest39458?location=US | 2021-06-25 07:50:10 | US       |
| turbottest20237 | https://bigquery.googleapis.com/bigquery/v2/projects/pikachu-aaa/jobs/turbottest20237?location=US | 2021-06-25 07:41:52 | US       |
| turbottest76164 | https://bigquery.googleapis.com/bigquery/v2/projects/pikachu-aaa/jobs/turbottest76164?location=US | 2021-06-25 07:46:57 | US       |
+-----------------+---------------------------------------------------------------------------------------------------+---------------------+----------+
```
</details>

### CODE REVIEW

File: docs/tables/gcp_bigquery_job.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+select
+  job_id,
+  self_link,
+  creation_time,
+  location
+from
+  gcp_bigquery_job;
+```
+
+### List jobs which are currently running
```
Comment (Subhajit97) [inline docs/tables/gcp_bigquery_job.md:19]:
```suggestion
### List running jobs
```

File: gcp/table_gcp_bigquery_job.go [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+				Description: "Name of the primary reservation assigned to this job.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Statistics.ReservationId"),
+			},
+			{
+				Name:        "self_link",
+				Description: "An URL that can be used to access the resource again.",
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getBigQueryJob,
+			},
```
Comment (Subhajit97) [inline gcp/table_gcp_bigquery_job.go:96]: Add this column after `state`.

File: gcp/table_gcp_bigquery_job.go [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+
+	return resp, nil
+}
+
+//// TRANSFORM FUNCTIONS
+
+func bigQueryJobAka(ctx context.Context, h *transform.TransformData) (interface{}, error) {
+	data := jobID(h.HydrateItem)
+	projectID := strings.Split(data, ":")[0]
+	id := strings.Split(data, ".")[1]
```
Comment (Subhajit97) [inline gcp/table_gcp_bigquery_job.go:314]: Can't we do the split operation inside `jobID()`?

**Review by Subhajit97 — APPROVED** [review]

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**